packages/payment-detection/src/near-info-retriever.ts

```diff
@@ -17,124 +21,45 @@ export class NearInfoRetriever {
     private paymentReference: string,
     private toAddress: string,
     private proxyContractName: string,
-    private procedureName: string,
     private eventName: PaymentTypes.EVENTS_NAMES,
     private network: string,
   ) {
     if (this.network !== 'aurora' && this.network !== 'aurora-testnet') {
       throw new Error('Near input data info-retriever only works with Near mainnet and testnet');
     }
-    this.nearWebSocketUrl = NEAR_WEB_SOCKET_URL;
+    if (this.network !== 'aurora') {
+      // FIXME: remove this check and implement testnet detection once aurora-testnet subgraphes are available
+      throw new Error(
+        'FIXME: getTransactionsFromNearSubGraph() only implemented for aurora mainnet',
+      );
+    }
+    this.network = this.network.replace('aurora', 'near');
+    if (this.proxyContractName !== 'requestnetwork.near') {
+      throw new Error(
+        `Proxy contract "${proxyContractName}" not supported by Near subgraph retriever`,
+      );
+    }
+    this.client = getTheGraphClient<'near'>(this.network);
   }
 
-  public async getTransferEvents(): Promise<PaymentTypes.ETHPaymentNetworkEvent[]> {
-    const events = await this.getTransactionsFromNearIndexerDatabase();
-    return events.map((transaction) => ({
-      amount: transaction.deposit,
+  public async getTransferEvents(): Promise<
+    PaymentTypes.IPaymentNetworkEvent<NearSubGraphPaymentEvent>[]
+  > {
+    const payments = await this.client.GetNearPayments({
+      reference: this.paymentReference,
+      to: this.toAddress,
+    });
+    return payments.payments.map((p) => ({
+      amount: p.amount,
       name: this.eventName,
       parameters: {
-        block: transaction.block,
-        confirmations: transaction.confirmations,
-        txHash: transaction.txHash,
+        block: p.block,
+        confirmations: p.block,
+        // Cf. FIXME#1 above
+        // txHash: transaction.txHash,
+        receiptId: p.receiptId ?? 'FIXME String!',
```

Comment:
I wait for the current graph indexing to be finished before I push a version where the receiptId cannot be undefined.